hatch9191/fit-check
# feat(FIT-15): basic email login and FE user sessions for next auth

### Overview

This PR adds the inital setup and proof of functionality for NextAuth in the app.

### Tasks

- [x] Adds the NextAuth v4 package (latest stable)
- [x] Uses the auto generated sign in page for testing
- [x] Create and sign in users with the form
- [x] Dynamically show login or sign up buttons depending on session

### Where to test

- Start the server locally
- Go to `http://localhost:3000/api/auth/signin` for auto generated page


### How to test

1. Use brand new user credentials for the sign in form
2. Should create a new user and push you to the main app
3. Use the Log out and Sign in buttons again with the same creds 

### Screenshots
<img width="450" alt="Screenshot 2024-07-15 at 09 59 59" src="https://github.com/user-attachments/assets/b9caba42-3edf-4074-bf09-4fc4b0d6e060">
<img width="450" alt="Screenshot 2024-07-15 at 09 59 54" src="https://github.com/user-attachments/assets/c7708ea2-90e9-48f8-bc88-e9ec9172b714">
<img width="494" alt="Screenshot 2024-07-15 at 09 59 12" src="https://github.com/user-attachments/assets/ee561ada-126a-4744-bd2b-552fcec77c16">
